src/include/storage/garbage_collector.h

```diff
@@ -64,6 +69,8 @@ class GarbageCollector {
   transaction::TransactionQueue txns_to_deallocate_;
   // queue of txns that need to be unlinked
   transaction::TransactionQueue txns_to_unlink_;
+  // queue of unexecuted deferred actions
+  transaction::ActionQueue deferred_actions_;
```

Comment:
Can we switch to a templated typedef instead?

Reply:
Can you elaborate?  Would you prefer that we no have the `ActionQueue` type and instead invoke the underlying template inline here?

Reply:
I meant that maybe we should have:

```
template <class Item>
using Queue = std::forward_list<Item>
...
transaction::Queue<Action *>
```

On second thought, it would be okay to simply explicitly say `forward_list` for all these. I think the typedef is leftover from back when we used `std::queue<transaction::TransactionContext *, std::list<transaction::TransactionContext *>>`

Reply:
Unfortunately, `forward_list` works for the GC containers because ordering doesn't really matter, but it would be painful for the `ActionQueue` because it is linear time to append (instead of constant time for `queue` (backed by either `deque` or `list`).

Reply:
Oh sorry. Misread that one.

We can probably just lose the typedef for your use case.

Reply:
The only reason I lean towards the typedef is that it makes it easier to transparently swap at the underlying data structure if it becomes a problem.

Reply:
We have 3 usages of this in our code base right now. I personally like to avoid typedefs unless the name really is just too long.

Typedef-ing does not provide information hiding. Code can still call whatever specialized function on the object they want. Wrappers are much better at doing that.

Reply:
Correct it doesn't provide information hiding, but it provides one place to change the underlying data structure with an API compatible replacement, instead of each and every usage point.

At this point I'm not sure what you think it should look like.  Would you prefer this be converted to a wrapper?

Reply:
I think for now just use the c++ typename directly.